### PR TITLE
improve tctl config loading ux

### LINKED ISSUE #25620: Using an identity file with `tctl` on a machine with `/etc/teleport.yaml` gives confusing error `tctl must be either used on the auth server or provided with the identity file via --identity flag`
When using `tctl --identity=./my-identity-file` on a host that has a Teleport configuration in the default directory (`/etc/teleport.yaml`), the following error is emitted:

```
ERROR: tctl must be either used on the auth server or provided with the identity file via --identity flag
```

We should either improve this error (as it does not explain how to rectify this situation), or when `--identity` is provided we should default to ignoring the Teleport configuration at `/etc/teleport.yaml`. When ignoring a Teleport configuration due to `--identity` being provided, we should emit a warning message that this is occurring.

**Workaround**

Set the `TELEPORT_CONFIG_FILE` environment variable to an empty string.

```sh
export TELEPORT_CONFIG_FILE=""
# Now you can use `tctl` without it trying to read the teleport configuration directory
```

### PR DESCRIPTION
**Pull request by GavinFrazar**
changelog: tctl will now ignore any configuration file if the auth_service section is disabled, and prefer loading credentials from a given identity file or tsh profile instead.

Fixes https://github.com/gravitational/teleport/issues/25620

This issue frequently trips customers (and me) up - see the linked issue and the issues linking to it.

This PR checks the config file that tctl finds and ignores it if the auth service section is not enabled.
That makes it far more friendly to use tctl on a host where a teleport agent is running (or even just configured).

I also updated the docs and error messages to hint towards tsh/identity file credential loading.

I'm not sure if this is a "breaking" change, but it seems that tctl only loads the config file to get an auth server address + credentials, both of which are provided by identity file (we already require the address flag be provided with an id file) and tsh profile already, so it seems fine to me.

### CODE REVIEW

**Comment by zmb3** [comment]
Nice!

File: tool/tctl/common/tctl.go [hunk 1]
```diff
@@ -148,11 +148,11 @@ func TryRun(commands []CLICommand, args []string) error {
 	app.Flag("debug", "Enable verbose logging to stderr").
 		Short('d').
 		BoolVar(&ccf.Debug)
-	app.Flag("config", fmt.Sprintf("Path to a configuration file [%v]. Can also be set via the %v environment variable.", defaults.ConfigFilePath, defaults.ConfigFileEnvar)).
+	app.Flag("config", fmt.Sprintf("Path to a configuration file [%v] for an auth service. Can also be set via the %v environment variable. Ignored if the auth_service is disabled.", defaults.ConfigFilePath, defaults.ConfigFileEnvar)).
```
Comment (ptgott) [inline tool/tctl/common/tctl.go:151]:
```suggestion
	app.Flag("config", fmt.Sprintf("Path to a configuration file [%v] for an Auth Service instance. Can also be set via the %v environment variable. Ignored if the auth_service is disabled.", defaults.ConfigFilePath, defaults.ConfigFileEnvar)).
```

For consistency with the docs

File: tool/tctl/common/tctl.go [hunk 2]
```diff
@@ -366,11 +379,11 @@ func ApplyConfig(ccf *GlobalCLIFlags, cfg *servicecfg.Config) (*authclient.Confi
 	if err != nil {
 		if errors.Is(err, fs.ErrNotExist) {
 			return nil, trace.Wrap(err, "Could not load Teleport host UUID file at %s. "+
-				"Please make sure that Teleport is up and running prior to using tctl.",
+				"Please make sure that a Teleport auth service is running on this host prior to using tctl or provide credentials by logging in with tsh first.",
```
Comment (ptgott) [inline tool/tctl/common/tctl.go:382]:
```suggestion
				"Please make sure that a Teleport Auth Service instance is running on this host prior to using tctl or provide credentials by logging in with tsh first.",
```
For consistency with the docs

File: tool/tctl/common/tctl.go [hunk 3]
```diff
@@ -380,7 +393,7 @@ func ApplyConfig(ccf *GlobalCLIFlags, cfg *servicecfg.Config) (*authclient.Confi
 		// The "admin" identity is not present? This means the tctl is running
 		// NOT on the auth server
 		if trace.IsNotFound(err) {
-			return nil, trace.AccessDenied("tctl must be either used on the auth server or provided with the identity file via --identity flag")
+			return nil, trace.AccessDenied("tctl must be used on an auth server host or provided with credentials by logging in with tsh first.")
```
Comment (ptgott) [inline tool/tctl/common/tctl.go:396]:
```suggestion
			return nil, trace.AccessDenied("tctl must be used on an Auth Service host or provided with credentials by logging in with tsh first.")
```

**Review by ptgott — APPROVED** [review]

**Review by espadolini — APPROVED** [review]

**Comment by GavinFrazar** [comment]
@gabrielcorado this needs G2

**Review by rosstimothy — APPROVED** [review]

**Comment by ravicious** [comment]
Would it be worthwhile to backport it to v15 as well? I think people are hitting this issue pretty often.

**Comment by webvictim** [comment]
Nice!

**Pull request merged**